### PR TITLE
[documentation] Feedback from 8.20 release process

### PR DESCRIPTION
**Pull request by proux01**
@Zimmi48 you're probably the ideal assignee here
Cc @silene if you see something I forgot

### CODE REVIEW

File: dev/doc/release-process.md
Comment (silene) [inline dev/doc/release-process.md:62]: Having this blob here does not make much sense to me. It is something that should be started much earlier in the process. Also, it is not specific to X.Y+rc, any X.Y.Z release needs its corresponding changelog PR to be merged in master too.

Comment (proux01) [reply dev/doc/release-process.md]: Indeed, I tried to improve that.

File: dev/doc/release-process.md
Comment (SkySkimmer) [inline dev/doc/release-process.md:25]: this links to "8.20 board" but how is the new project created? Is it supposed to be a separate project from the current "release management" project (projects/11) or is it supposed to be some sub-thing of the "release management" project?

Comment (proux01) [reply dev/doc/release-process.md:25]: Good question, no idea, @Zimmi48 ?

Comment (Zimmi48) [reply dev/doc/release-process.md:25]: Sorry, I should not have waited so long to document this. The idea is to use a single release management project from now on.

The process is the following:

- Create a milestone.
- Add a description saying to which branch(es) its merged PRs should get backported. A simple case is `@coqbot: backport to v8.20 (move rejected PRs to: https://github.com/coq/coq/milestone/60).` A more complex case would look like `@coqbot: backport to v8.19 (move rejected PRs to: <url of current milestone for 8.20>); backport to v8.20 (move rejected PRs to: <URL of current milestone for 8.21>); ...`.
- As soon as a PR is merged in this milestone, the project field for its backporting status will be automatically created by coqbot (with standard values "Request inclusion", "Shipped" and "Rejected").
- You can use pre-existing views in the project to track PR backporting or create new views for new major releases (on the model of previous releases). This could also be automatically done by coqbot, but it isn't the case yet.
- If a PR status is changed to rejected without the milestone being changed first, then coqbot posts a comment to the PR and automatically changes the milestone to the "move rejected PRs to" milestone value.

File: dev/doc/release-process.md
Comment (Zimmi48) [inline dev/doc/release-process.md:5]: It took me a while to understand this parenthesis. Maybe this will make it clearer.

```suggestion
- [ ] Create a new issue to track the release process where you can copy-paste the present checklist from `dev/doc/release-process.md` (we recommend duplicating the "For each release" section for X.Y+rc1 and for X.Y.0, removing the useless entries in each copy).
```

File: dev/doc/release-process.md
Comment (Zimmi48) [inline dev/doc/release-process.md:41]:
```suggestion
Be sure the PR is not draft for better visibility and ask everyone in the dev team to contribute the main features and breaking changes sections before the final release.
```

File: dev/doc/release-process.md
Comment (Zimmi48) [inline dev/doc/release-process.md:59]: AFAIU, SerAPI is no longer maintained and thus no longer included in Docker-Coq images, therefore, this line can be removed. (Cf. https://github.com/coq-community/manifesto/issues/160)

File: dev/doc/release-process.md
Comment (Zimmi48) [inline dev/doc/release-process.md:61]:
```suggestion
- [ ] Cc `@erikmd` to ensure that the necessary configuration is ready to release the Docker images in [`coqorg/coq`](https://hub.docker.com/r/coqorg/coq) (gathering `coq` and `coq-bignums` opam packages).
```

File: dev/doc/release-process.md
Comment (Zimmi48) [inline dev/doc/release-process.md:62]:
```suggestion
- [ ] For X.Y+rc1, send an e-mail on coqdev + coq-club + the Coq announcement category on Discourse (coqdev@inria.fr + coq-club@inria.fr + coq+announcements@discoursemail.com) announcing that the tag has been set so that package managers can start preparing package updates and library authors can safely start preparing compatible releases.
```

File: dev/doc/release-process.md
Comment (proux01) [reply dev/doc/release-process.md:59]: @erikmd do you confirm?

File: dev/doc/release-process.md
Comment (proux01) [reply dev/doc/release-process.md]: Thanks @Zimmi48 I tried to translate that in the PR, feel free to criticize

File: dev/doc/release-process.md
Comment (proux01) [reply dev/doc/release-process.md]: @erikmd done that

File: dev/doc/release-process.md
Comment (Zimmi48) [inline dev/doc/release-process.md:89]: Probably the whole section "## This is now delegated to the platform maintainers" should be removed, because it is clearly not maintained by the people involved and is becoming seriously outdated.

File: dev/doc/release-process.md
Comment (Zimmi48) [inline dev/doc/release-process.md:64]: For consistency with the rest of the document:

```suggestion
- [ ] For X.Y.0, once opam and Nix packages are ready (and ideally Docker images), announce the release to Coq-Club and Discourse (coq-club@inria.fr + coq+announcements@discoursemail.com).
```

File: dev/doc/release-process.md
Comment (Zimmi48) [inline dev/doc/release-process.md:25]: Maybe the part about the milestone and the backporting project should be moved above the part about the tag because, while the tag can wait without creating trouble, if the milestone description is not set as soon as the new branch is created, there is a risk of missing out some PRs that would be merged in the meantime.

File: dev/doc/release-process.md
Comment (Zimmi48) [inline dev/doc/release-process.md:26]: Cosmetic:

```suggestion
- [ ] Monitor the [Release management project](https://github.com/orgs/coq/projects/11) in which coqbot will keep track of PRs to be backported (according to the previous command)
```

**Review by Zimmi48 — APPROVED** [review]
Thanks :pray: This looks very good to me!

File: dev/doc/release-process.md
Comment (silene) [reply dev/doc/release-process.md:25]: I don't understand the part about having several backport directives. If we also write `backport to v8.19` inside the description of 8.20.1 milestone, this is bound to drive a lot of people crazy.

Comment (proux01) [reply dev/doc/release-process.md:25]: I think the use case is rather the contrary: something tagged with milestone 8.19.3 should: backport to 8.19.3; backport to 8.20.1
@Zimmi48 is it the use case you have in mind, if yes, shouldn't we rather recommend when creating a milestone to add the "backport to <new_milestone>" to all still open milestone for minor versions of previous releases?

Comment (silene) [reply dev/doc/release-process.md:25]: Oh, that makes a lot more sense when you word it that way!

Comment (Zimmi48) [reply dev/doc/release-process.md:25]: Indeed, this has happened before that this was needed. And yes, @proux01 I think adding the recommendation you mentioned would be worthwhile. (There have also been cases before where milestones like this were created, but this step was forgotten, and because of this some backports were missed).

Comment (proux01) [reply dev/doc/release-process.md]: @Zimmi48 done, if you're happy with it, feel free to assign and merge

**Comment by Zimmi48** [comment]
@coqbot merge now

**Pull request merged**
